snuba/query/snql/parser.py

```diff
@@ -734,6 +735,44 @@ def parse_snql_query_initial(
     return parsed
 
 
+def _qualify_columns(query: Union[CompositeQuery[QueryEntity], LogicalQuery]) -> None:
+    """
+    All columns in a join query should be qualified with the entity alias, e.g. e.event_id
+    Take those aliases and put them in the table name. This has to be done in a post
+    process since we need to have all the aliases from the join clause.
+    """
+
+    from_clause = query.get_from_clause()
+    if isinstance(query, LogicalQuery) or not isinstance(from_clause, JoinClause):
```

Comment:
You should not need to check the query is not a LogicalQuery if you already check that the from_clause is a JoinClause.

Reply:
You're right, I thought mypy might complain.